awx/main/analytics/broadcast_websocket.py

```diff
@@ -0,0 +1,159 @@
+import datetime
+import asyncio
+import logging
+import aioredis
+import redis
+
+from prometheus_client import (
+    generate_latest,
+    Gauge,
+    Counter,
+    Enum,
+    CollectorRegistry,
+)
+
+from django.conf import settings
+
+
+BROADCAST_WEBSOCKET_REDIS_KEY_NAME = 'broadcast_websocket_stats'
+
+
+logger = logging.getLogger('awx.main.analytics.broadcast_websocket')
+
+
+def dt_to_seconds(dt):
+    return int((dt - datetime.datetime(1970,1,1)).total_seconds())
+
+
+def now_seconds():
+    return dt_to_seconds(datetime.datetime.now())
+
+
+# Second granularity; Per-minute
+class FixedSlidingWindow():
+    def __init__(self, start_time=None):
+        self.buckets = dict()
+        self.start_time = start_time or now_seconds()
+
+    def cleanup(self, now_bucket=now_seconds()):
+        if self.start_time + 60 >= now_bucket:
+            self.start_time = now_bucket - 60 + 1
+
+            # Delete old entries
+            for k,v in self.buckets.items():
+                if k < self.start_time:
+                    del self.buckets[k]
+
+    def record(self, ts=datetime.datetime.now()):
```

Comment:
This will call `datetime.now()` once when the module is initialized and reuse the result for all instances of `FixedSlidingWindow`.

Reply:
Yes, seems like if you specify the default in this way, you're gonna have a default `ts` that's always the start time of when this module is imported (when the process starts).